### PR TITLE
bookkeeper-daemon script supports restart command

### PR DESCRIPTION
**Pull request by gaozhangmin**
bookkeeper-daemon script supports restart command

### CODE REVIEW

**Comment by gaozhangmin** [comment]
rerun failure checks

File: bin/bookkeeper-daemon.sh [hunk 1]
```diff
@@ -108,74 +108,101 @@ rotate_out_log ()
...
+        stop
+      fi
+      if [ "$?" == 0 ]
+      then
+         sleep 3
+         paramaters="$*"
+         startParamaters=${paramaters//-force/}
+         start "$startParamaters"
+      else
+         echo "WARNNING :  $command failed restart, for $command is not stopped completely."
```
Comment (nicoloboschi) [inline bin/bookkeeper-daemon.sh:203]:
```suggestion
         echo "WARNING : $command restart failed, $command is not stopped completely."
```

File: bin/bookkeeper-daemon.sh [hunk 2]
```diff
@@ -108,74 +108,101 @@ rotate_out_log ()
...
+      if [[ "$forceStopFlag" != "" ]]
+      then
+        stop "-force"
+      else
+        stop
+      fi
+      if [ "$?" == 0 ]
+      then
+         sleep 3
+         paramaters="$*"
```
Comment (nicoloboschi) [inline bin/bookkeeper-daemon.sh:199]: typo: parameters

File: bin/bookkeeper-daemon.sh [hunk 3]
```diff
@@ -108,74 +108,101 @@ rotate_out_log ()
...
+         startParamaters=${paramaters//-force/}
```
Comment (nicoloboschi) [inline bin/bookkeeper-daemon.sh:200]: just a nit: if one of the command we have to start here accepts a flag like `-force-xxx` it will be blanked as well

**Review by nicoloboschi — CHANGES_REQUESTED** [review]

**Comment by gaozhangmin** [comment]
rerun failure checks

**Review by nicoloboschi — APPROVED** [review]

**Review by zymap — CHANGES_REQUESTED** [review]
Please check the indent, looks like sometimes they are two spaces, sometimes they are four spaces.

**Comment by gaozhangmin** [comment]
> Please check the indent, looks like sometimes they are two spaces, sometimes they are four spaces.

@zymap  done.

File: bin/bookkeeper-daemon.sh [hunk 4]
```diff
@@ -108,9 +108,9 @@ rotate_out_log ()
 
 mkdir -p "$BOOKIE_LOG_DIR"
 
-case $startStop in
-  (start)
-    if [ -f $pid_file ]; then
+start()
+{
+  if [ -f $pid_file ]; then
```
Comment (zymap) [inline bin/bookkeeper-daemon.sh:113]: line 114 to 130 still have the indent issue. They need to move forward.

File: bin/bookkeeper-daemon.sh [hunk 5]
```diff
@@ -128,10 +128,11 @@ case $startStop in
...
       exit 1
     fi
-    ;;
+}
 
-  (stop)
-    if [ -f $pid_file ]; then
+stop()
+{
+  if [ -f $pid_file ]; then
```
Comment (zymap) [inline bin/bookkeeper-daemon.sh:135]: same issue with above.

**Review by zymap — APPROVED** [review]

**Pull request merged**